Feature/hocs 2907 compose up all (#4)
UKHomeOffice/hocs: docker/README.md

# docker-compose

This directory contains configuration for Docker Compose, a quick way to get a
local HOCS environment on your computer. 

If this is the first time you run it, Compose will download the latest images; this could

> take fifteen minutes, depending on your connection. To relieve pressure on your machines resources, 

Reviewer: ```suggestion
take fifteen minutes, depending on your connection. To relieve pressure on your machine&rsquo;s resources, 
```

Author: done

$ docker-compose up
```

## Use of .env file
The docker-compose file has variables for the image tags. This allows the use of a ``.env`` file.

> By default, ``branch-main`` will be used. The following is an example of a file overriding some services:

Reviewer: ```suggestion
By default, ``latest`` will be used. The following is an example of a file overriding some services:
```

Author: good spot, done

> By default, ``branch-main`` will be used. The following is an example of a file overriding some services :

Reviewer: ```suggestion
By default, ``branch-main`` will be used. The following is an example of a file overriding some services:
```

It'd be better to default to `latest` as that's, well, the default behaviour.

Author: done

```shell
FRONTEND_TAG=branch-epic_HOCS-COMP
CASEWORK_TAG=branch-epic_HOCS-COMP
WORKFLOW_TAG=branch-epic_HOCS-COMP
CASE_CREATOR_TAG=branch-epic_HOCS-COMP
INFO_TAG=branch-epic_HOCS-COMP
HOCS_DATA_TAG=branch-feature_HOCS-2907-compose-up-all
HOCS_DATA_REPO=hocs-data
HOCS_DATA_ELASTIC_REPO=hocs-data-elastic
```

> The `HOCS_DATA_REPO` can be one of :

Reviewer: Colons don't generally have leading spaces. Don't make me dust off my Chicago Manual of Style.
```suggestion
The `HOCS_DATA_REPO` can be one of:
```

Author: done

* hocs-data
* hocs-data-wcs

> The `HOCS_DATA_ELASTIC_REPO` can be one of :

Reviewer: ```suggestion
The `HOCS_DATA_ELASTIC_REPO` can be one of:
```

Author: done

* hocs-data-elastic
* hocs-data-wcs-elastic

## Data migration

> The info-service doesn&rsquo;t come with its schema, it relies on a data_migration step in the docker-compose file.

Author: Changed

The data_migration step loads the info schema using flyway scripts.

> To select hocs-data pr hocs-data-wcs set the variable ``HOCS_DATA_REPO`` in your ``.env`` file to either :

Reviewer: ```suggestion
To select hocs-data or hocs-data-wcs set the variable ``HOCS_DATA_REPO`` in your ``.env`` file to either:
```

Author: done

* hocs-data (default)
* hocs-data-wcs

The docker image for the data is stored in AWS ECR, to pull it you will need to set up
an AWS profile and then run the following:
```console

> $ $(aws ecr get-login --no-include-email --profile acp-ecr)

Author: That gets the long login string and applies it.

```

To be able to pull the image you need to be in authorised to do so.
This is in the ACP Hub in the Docker repos section.

> The AWS credentials are stored in the hub, under your "Connected Identities" section.

Reviewer: ```suggestion
AWS credentials are stored in the ACP hub, under your &ldquo;Connected Identities&rdquo; section.
```

Author: done

For more serious problems you might need to tear down the environment with
`docker-compose down` and start again.

### 403 on Frontend

The frontend container doesn&rsquo;t handle authentication, as it is handled
by a proxy before it hits the app. You&rsquo;ll need to set authentication
headers yourself using (for example) a browser extension. Ask an existing
developer for their copy. This might manifest itself as a browser timeout:
check stderr before assuming something else is wrong.

### Manually applying the Search indexes

The search service needs elastic indexes applying. Without them, you will see continuous search errors.

> The search indexes are contained in either hocs-data or hocs-data-wcs. To apply the index, do th following:

Author: corrected
